cmd/snap-update-ns/user.go

```diff
@@ -37,16 +37,24 @@ type UserProfileUpdateContext struct {
 }
 
 // NewUserProfileUpdateContext returns encapsulated information for performing a per-user mount namespace update.
-func NewUserProfileUpdateContext(instanceName string, uid int) *UserProfileUpdateContext {
+func NewUserProfileUpdateContext(instanceName string, fromSnapConfine bool, uid int) *UserProfileUpdateContext {
```

Comment:
I'd rather we had two initialisers, than one with a boolean flag. OTOH if there are going to be more ways, an options struct maybe?

Reply:
This is all that we have and we only invoke it in one place (never with a literal boolean) outside of test code. Once snap-boostrap-ns comes around this will be easier as the option will go away entirely. In that world snap-update-ns is always invoked by snapd and snap-boostrap-ns is always invoked by snap-confine. Then the options are gone :)